pkg/crc/api/api_http.go

```diff
@@ -85,9 +87,39 @@ func NewMux(config crcConfig.Storage, machine machine.Client, logger Logger, tel
 		}
 		sendResponse(w, handler.UploadTelemetry(data))
 	})
+
+	mux.HandleFunc("/pull-secret", pullSecretHandler(config))
+
 	return mux
 }
 
+func pullSecretHandler(config crcConfig.Storage) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This could've gone into the handlers file :) (time to retire the binary protocol :p) and we allow all the http request types for this..

Reply:
yes.. the handler is not really meant to deal with status code. It was really easier to do that here. I hope to merge most structs in api package when the binary server will be removed.